### DISCUSSION TITLE
Asking fabric to summarize the contents of a txt file

### DISCUSSION THREAD

**Discussion by duttad** [danielmiessler/Fabric#386] [Q&A]
I must have missed something, but the -t flag doesn't seem to work with files.  The error message is below. It's running on Ubuntu 22.04 (WSL-2)
```
$ fabric -t section_25.txt -o section_25_fabric_analyze_threat_report.txt -p analyze_threat_report
... RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
I'm sorry, but I can't process or generate content based on an unspecified input like "section_25.txt" without the actual content of the file being provided. Please provide the text or details you'd like analyzed or summarized.
```

**Comment by duttad** [comment]
`$ cat section_25.txt | fabric -o section_25_fabric_analyze_threat_report.txt -p analyze_threat_report` works. So I am guessing it's a permissions issue on WSL.

**Reply by scottrkahler** [reply]
I need help with plain text files too.

**Comment by pboin** [comment]
I think that your '-t' flag here is passing the literal text to the model.  In other words you asked to analyze_threat_report against the string "section_25.txt" -- which makes no sense.  Read your error message that way and see if that makes sense.

**Comment by ulagu** [comment]
Try this: it will read the text file with cat, call the fabric pattern, then save the output to another text file. 

Cat section_25.txt | fabric -p analyze_threat_report | tee section_25_fabric_analyze_threat_report.txt
